project44/manifest-ui
# feat(themes): add Now theme

# Description

DES-15 | Assist teams from migrating from Now to 2.0 themed styles

We're going back to SHARP AND POINTY 🔪 

There are three main value changes that distinguish this theme from the default:

1. color
2. radius
3. shadows

![nowtheme](https://user-images.githubusercontent.com/88858003/153524428-45bd11ea-db07-4885-be6a-44a80b851202.jpg)

# 🔍  How to test this change locally

1. You can switch out the default theme for the nowTheme by implementing this diff. 
1. Run `yarn && yarn build && yarn dev`
1. Confirm changes by selecting the primary button and inspecting that the radius is, indeed, 0.

```
diff --git a/.storybook/preview.tsx b/.storybook/preview.tsx
index 3f54322..2f2d7d9 100644
--- a/.storybook/preview.tsx
+++ b/.storybook/preview.tsx
@@ -1,10 +1,11 @@
 import * as React from 'react';
 import { Provider } from '@manifest-ui/provider';
 import { Story } from '@storybook/react';
+import { nowTheme } from '../packages/themes';

 export const decorators = [
   (StoryFn: Story) => (
-    <Provider>
+    <Provider theme={nowTheme}>
       <StoryFn />
     </Provider>
   ),
diff --git a/packages/themes/src/index.ts b/packages/themes/src/index.ts
index acced89..6ec8f3b 100644
--- a/packages/themes/src/index.ts
+++ b/packages/themes/src/index.ts
@@ -1 +1,2 @@
 export * from './default';
+export * from './nowTheme';
```

# 🔮  Outstanding Questions

* I removed all unused keys from the theme since the fallback is the default. Is this right? Should I go back and copy/pasta back those key/value pairs?
* No strong opinions on the naming of the file and theme. If you have a stronger suggestion, I'm game.
